### PR TITLE
Add ability to configure the S3 endpoint

### PR DESCRIPTION
**Pull request by dfunckt**
This is useful when deploying against an S3-like service for storage, eg. Minio.

Change-type: minor

### CODE REVIEW

File: config/confd_env_backend/templates/docker-registry.yml.tmpl [hunk 1]
```diff
@@ -23,6 +23,9 @@ storage:
     accesskey: {{getenv "REGISTRY2_S3_KEY"}}
     secretkey: {{getenv "REGISTRY2_S3_SECRET"}}
     region: {{getenv "COMMON_REGION"}}
+{{if exists "REGISTRY2_S3_REGION_ENDPOINT"}}
+    regionendpoint: {{getv "REGISTRY2_S3_REGION_ENDPOINT"}}
```
Comment (hedss) [inline config/confd_env_backend/templates/docker-registry.yml.tmpl]:
```suggestion
    regionendpoint: {{getenv "REGISTRY2_S3_REGION_ENDPOINT"}}
```

File: config/confd/templates/docker-registry.yml.tmpl [hunk 2]
```diff
@@ -38,6 +41,3 @@ redis:
   addr: {{getv "/resin/registry2/cache-addr" "127.0.0.1:6379"}}
   db: {{getv "/resin/registry2/cache-db" "0"}}
 {{end}}
-
-redirect:
```
Comment (Page-) [inline config/confd/templates/docker-registry.yml.tmpl]: Why is this removed?

Comment (dfunckt) [reply config/confd/templates/docker-registry.yml.tmpl]: It never worked as intended — it’s specified in the wrong place and registry just ignores it.
